pkg/gadgets/run/types/factory.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"reflect"
+	"unsafe"
+
+	"golang.org/x/exp/constraints"
+)
+
+type EventFactory struct {
+	lastOffset uintptr
+	lastIndex  int
```

Comment:
The name is confusing me. Isn't `nextIndex`/`nextFreeIndex` here better? In the following code the usage seems to be that this indicates exactly that.

Or change the usage that this is actually the last valid index